docs/step-by-step/kubeapps-on-tkg/step-1.md

```diff
@@ -0,0 +1,127 @@
+
+6. Enter a name and description for the OAuth app. For the moment, enter the value `https://localhost/oauth2/callback` in the _Redirect URIs_ field (this will be updated after Kubeapps is installed).
+
+   ![Add name and description](./img/csp-oauth-new-details-general.png)
+
+7. Leave the rest of the options at their default values.
+8. Tick the _OpenID_ checkbox and click the _Create_ button.
+
+   ![OpenID and create](./img/csp-oauth-new-details-scopes.png)
+
+The CSP Console displays a success screen with an auto-generated application ID and secret. Click the _Download JSON_ link to download these values.
+
+![Retrieve app id and secret](./img/csp-oauth-new-secrets.png)
+
+> **NOTE**: Store this file carefully as it contains important credentials which will be needed when configuring Pinniped and Kubeapps.
+
+> **TIP**: Any OIDC-compliant provider should expose a `.well-known/openid-configuration` ([example](https://console.cloud.vmware.com/csp/gateway/am/api/.well-known/openid-configuration)) endpoint where you can find other useful and required information. This endpoint allows using the base URL to discover the rest of the URLs (`authorization`, `token`, `end session`, `jwks` and `issuer`) automatically. For CSP, the endpoint is `https://console-stg.cloud.vmware.com/csp/gateway/am/api`.
+
+At this point, an OAuth2 application is configured.
+
+#### Configure Pinniped to Trust the OIDC Provider
+
+Once the OIDC provider has been fully configured, the next step is to configure Pinniped to trust this provider. This implies that a successful authentication with the OIDC provider results in authentication with the TKG cluster.
+
+Since Pinniped manages this process, the only requirement is to a _JWTAuthenticator_ CustomResource in the cluster. To do so, follow the steps below:
+
+1. Create a file named `kubeapps-jwt-authenticator.yaml` with the following content. Replace the placeholders as follows:
+
+- Replace the `OIDC-ISSUER-URL` with the issuer URL of the OIDC provider. For CSP it is `https://console-stg.cloud.vmware.com/csp/gateway/am/api`.
+- Replace `CLIENT-ID` with the application ID obtained from the JSON file in the previous step.
+
+```yaml
+---
+apiVersion: authentication.concierge.pinniped.tmc.cloud.vmware.com/v1alpha1
+kind: JWTAuthenticator
+metadata:
+  name: kubeapps-jwt-authenticator
+spec:
+  issuer: OIDC-ISSUER-URL
+  audience: CLIENT-ID
+  claims:
+    username: email
+#   tls:
+#     certificateAuthorityData: LS0t... # optional base64 CA data if using a self-signed certificate
+```
+
+The `name` field specifies the name of the _JWTAuthenticator_ resource, which will be required in the next step.
+
+> **NOTE**: Ignore the `tls` section of the configuration shown above unless the OIDC uses a self-signed certificate. If it does, follow [these additional steps](https://github.com/kubeapps/kubeapps/blob/master/docs/user/using-an-OIDC-provider-with-pinniped.md#pinniped-not-trusting-your-oidc-provider).
```

Comment:
Hah - or if you happen to be running the TMC installed version which (last I checked) had stripped the trusted certs from the image ;P. But it was going to be fixed anyway.

Reply:
I recall having seen a commit reverting it, but I don't if it's already landed or not (but it's not worthy to add it in the docs)